pkgs/tools/graphics/pngloss/default.nix

```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchgit, cmake, lib, pkgs }:
+stdenv.mkDerivation rec {
+  name = "pngloss";
+  src = fetchgit {
+    url = "https://github.com/foobaz/pngloss.git";
+    rev = "559f09437e1c797a1eaf08dfdcddd9b082f0e09c";
+    sha256 = "sha256-dqrrzbLu4znyWOlTDIf56O3efxszetiP+CdFiy2PBd4=";
+  };
+  buildInputs = with pkgs;[
+    libpng
+    pkg-config
+    zlib
+  ];
+  builder = ./builder.sh;
+  
+  meta = with lib;{
+    description = "Lossy compression of PNG images";
+    homepage = "https://github.com/foobaz/pngloss";
+    license = licenses.mit;
+    platforms = platforms.all;
+  };
+}
```

Comment:
Thank you for your review. To make this PR, I referred to some unofficial manuals that explains how to *build a project with Makefile using nix, so I didn't understand much of the philosophy.
> When you are fetching code from github, it is preferred to use fetchFromGitHub instead of fetchzip. It has an easier and more native interface.

I see. It looks far neater than the way I did. 

> The programs/tools go into nativeBuildInputs, because when you cross compile they still have to be native to your own architecture, not the one you're compiling to.

pkg-config was not installed on my darwin environment, and I didn't know where to put that in. Now I understand it...

>Then a bigger mistake, instead of supplying your own build script mkDerivation can do all of it for you already.
In your build script you still seem to copy the binary over manually, so I included that in the idiomatic way by using the installPhase.

I also didn't know that mkDerivation does all the stuff automagically...

I've learned a lot of things from this review. Again, thank you so much.